test/spec/services/angular-apimock.js

```diff
@@ -377,6 +376,45 @@ describe('Service: apiMock', function () {
 				});
 
 			});
+			
+			describe('enable query params', function () {
+
+				beforeEach(function() {
+					apiMockProvider.config({stripQueries: false});
+					// Set global flag to mock
+					$location.search('apiMock', true);
+				});
+
+				afterEach(function() {
+					apiMockProvider.config({stripQueries: true});
+				});
+
+				it('should NOT ignore query objects in request URL (path has /?)', function () {
+					defaultRequest.url = '/api/pokemon/?name=pikachu';
+					defaultExpectPath = '/mock_data/pokemon/name=pikachu.get.json';
+					expectHttpSuccess();
+				});
+
+				it('should NOT ignore query objects in request URL (path has only ?)', function () {
+					defaultRequest.url = '/api/pokemon?name=pikachu';
+					defaultExpectPath = '/mock_data/pokemon/name=pikachu.get.json';
+					expectHttpSuccess();
+				});
+
+				it('should NOT ignore query objects in request OBJECT', function () {
+					defaultRequest.url = '/api/pokemon';
+					defaultRequest.params = {'name': 'pikachu', 'strength': 'electricity'};
+					defaultExpectPath = '/mock_data/pokemon/name=pikachu&strength=electricity.get.json';
+					expectHttpSuccess();
+				});
+
+				it('should sort query objects in alphabetical order', function () {
+					defaultRequest.url = '/api/pokemon';
+					defaultRequest.params = {'strength': 'electricity', 'name': 'pikachu'};
```

Comment:
Since you can also pass uri parameter values in the params object I don't think we just convert all params to a querystring.  We won't know which are query params and which are uri params.

Can we change the intercept point to intercept after Angular has already constructed the final url?  That would be much simpler.